Cython/Compiler/Symtab.py

```diff
@@ -1249,24 +1268,30 @@ def lookup_submodule(self, name):
             module = module.lookup_submodule(submodule)
         return module
 
-    def add_include_file(self, filename, late=False):
-        if filename in self.python_include_files:
-            return
-        # Possibly, the same include appears both as early and as late
-        # include. We'll deal with this at code generation time.
-        if late:
-            incs = self.include_files_late
-        else:
-            incs = self.include_files_early
-        if filename not in incs:
-            incs.append(filename)
+    def add_include_file(self, filename, verbatim_include=None, late=False):
+        """
+        Add `filename` as include file. Add `verbatim_include` as
+        verbatim text in the C file.
+        Both `filename` and `verbatim_include` can be `None` or empty.
+        """
+        # If filename is None or empty, this is just a template for
+        # verbatim_include below.
+        inc = Code.IncludeCode(filename, late=late)
+        if filename:
+            inc = inc.dict_update(self.c_includes, filename)
+        if verbatim_include:
+            # Add verbatim_include with the same location and order such
+            # that it comes right after the include "inc" in the
+            # generated file.
+            verb = Code.IncludeCode(verbatim_include, verbatim=True,
+                    location=inc.location, order=inc.order)
```

Comment:
Since both `location` and `order` can change later, the values used here seem arbitrary.